### DISCUSSION TITLE
How to integrate the code actions with telescope?

### DISCUSSION THREAD

**Discussion by mamazu** [phpactor/phpactor#2649] [Q&A]
In the docs there is a section about integrating with telescope `doc/lsp/vim-lsp.rst` but this looks like it's not the preferred way anymore (seeing that this is using commands).

My specific problem is that when using the code actions (especially the unresolved import and override method ones) the confirmation thing can get quite long and I would like to search them

**Comment by dantleech** [comment]
I mean, doesn't Telescope integrate natively?

**Reply by mamazu** [reply]
Not for me: 
```lua
    vim.api.nvim_buf_set_keymap(bufnr, 'n', '<Leader>f', '<cmd>lua vim.lsp.buf.code_action()<CR>', opts)
```
That's the line that's triggering it. Which by default uses the nvim selection mechanism.

![image](https://github.com/phpactor/phpactor/assets/14860264/443c4928-91ed-4835-b1f1-c6296beaa1e2)

**Reply by mamazu** [reply]
However oddly, like "find references" opens the resulting location list in telescope.
